api/capi/src/nnstreamer-capi-single-new.c

```diff
@@ -0,0 +1,500 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @file nnstreamer-capi-single-new.c
+ * @date 29 Aug 2019
+ * @brief NNStreamer/Single C-API Wrapper.
+ *        This allows to invoke individual input frame with NNStreamer.
+ * @see	https://github.com/nnsuite/nnstreamer
+ * @author MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include <string.h>
+
+#include <nnstreamer-single.h>
+#include <nnstreamer-capi-private.h>
+#include <nnstreamer_plugin_api.h>
+#include <nnstreamer/tensor_filter/tensor_filter.h>
+
+#include "tensor_filter_single.h"
+
+/* ML single api data structure for handle */
+typedef struct
+{
+  GTensorFilterSingle *filter;
+  ml_tensors_info_s in_info;
+  ml_tensors_info_s out_info;
+} ml_single;
+
+/**
+ * @brief Set the info for input/output tensors
+ */
+static void
+ml_single_set_inout_tensors_info (GObject *object,
+    const gchar *prefix, ml_tensors_info_s *tensors_info)
+{
+  GstTensorsInfo info;
+  gchar *str_dim, *str_type, *str_name;
+  gchar *str_type_name, *str_name_name;
+
+  ml_tensors_info_copy_from_ml (&info, tensors_info);
+
+  /* Set input option */
+  str_dim = gst_tensors_info_get_dimensions_string (&info);
+  str_type = gst_tensors_info_get_types_string (&info);
+  str_name = gst_tensors_info_get_names_string (&info);
+
+  str_type_name = g_strdup_printf("%s%s", prefix, "type");
+  str_name_name = g_strdup_printf("%s%s", prefix, "name");
+
+  g_object_set (object, prefix, str_dim,
+      str_type_name, str_type,
+      str_name_name, str_name, NULL);
+
+  g_free (str_type_name);
+  g_free (str_name_name);
+  g_free (str_dim);
+  g_free (str_type);
+  g_free (str_name);
+
+  gst_tensors_info_free (&info);
+}
+
+/**
+ * @brief Check the availability of the nnfw type and model
+ */
+static int
+ml_single_check_nnfw (const char *model, ml_nnfw_type_e *nnfw)
+{
+  gchar *path_down;
+  int status = ML_ERROR_NONE;
+
+  /* Check file extention. */
+  path_down = g_ascii_strdown (model, -1);
+
+  switch (*nnfw) {
+    case ML_NNFW_TYPE_ANY:
+      if (g_str_has_suffix (path_down, ".tflite")) {
+        ml_logi ("The given model [%s] is supposed a tensorflow-lite model.",
+            model);
+        *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
+      } else if (g_str_has_suffix (path_down, ".pb")) {
+        ml_logi ("The given model [%s] is supposed a tensorflow model.", model);
+        *nnfw = ML_NNFW_TYPE_TENSORFLOW;
+      } else if (!g_str_has_suffix (path_down, ".so")) {
```

Comment:
line 99 and 109
let's consider to use NNSTREAMER_SO_FILE_EXTENSION for so-file extension.

Reply:
Updated.